### PR TITLE
Add PortManager methods

### PR DESCRIPTION
**Pull request by cljohnso**
This commit adds PortManager.getPortRef and PortManager.isReservablePort
methods that may be useful in certain usage scenarios.  It also adds an
PortRef.isClosed method.

### CODE REVIEW

File: port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java [hunk 1]
```diff
@@ -159,6 +164,55 @@ private PortManager() {
...
+   * @param port the port number for which the {@code PortRef} is returned
+   * @return an {@code Optional} of the {@code PortRef} for {@code port} if the
+   *      {@code PortRef} is both strongly reachable and not closed;
+   *      {@code Optional.empty} if the {@code PortRef} for {@code port} is either
+   *      not strongly reachable or closed
+   * @throws IllegalArgumentException if {@code port} is not between 0 and
+   *    {@value #MAXIMUM_PORT_NUMBER} (inclusive) or is not a reservable port
+   */
+  public synchronized Optional<PortRef> getPortRef(int port) {
+    if (port < 0 || port > MAXIMUM_PORT_NUMBER) {
```
Comment (chrisdennis) [inline port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java:199]: Couldn't these 6 lines be replaced with a call to `isReservablePort(port)`?

File: port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java [hunk 2]
```diff
@@ -159,6 +164,55 @@ private PortManager() {
...
+      throw new IllegalArgumentException("Port " + port + " is not a valid port number");
+    }
+    if (restrictedPorts.get(port)) {
+      throw new IllegalArgumentException("Port " + port + " is not reservable");
+    }
+
+    cleanReleasedPorts();
```
Comment (chrisdennis) [inline port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java:206]: I think you can drop these 4 lines.

**Review by chrisdennis — APPROVED** [review]
Approved, change at your discretion.

File: port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java (hunk 1 above)
Comment (cljohnso) [reply port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java:199]: Almost.  Modifying.

File: port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java (hunk 2 above)
Comment (cljohnso) [reply port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java:206]: I agree we can get rid of the `if` but I'm pretty sure the call to `cleanReleasedPorts` needs to be retained.

**Pull request merged**